modin/core/dataframe/pandas/dataframe/dataframe.py

```diff
@@ -320,6 +335,20 @@ def set_columns_cache(self, columns):
         else:
             self._columns_cache = ModinIndex(columns)
 
+    def set_axis_cache(self, value, axis=0):
```

Comment:
Why is this additional method/wrapper if we already have what we need?

Reply:
it's really inconvenient to if-else every time when we need to set a cache depending on an axis:
```python
# old version
if axis == 0:
    self.set_index_cache(self.copy_index_cache())
else:
    self.set_columns_cache(self.copy_columns_cache())

# new version
self.set_axis_cache(self.copy_axis_cache(axis), axis)
```

Reply:
why? IMO all of these methods are useful and can be in a public namespace.

We have `.index`, `.columns`, and `.axes` and they're all public which doesn't cause any problems

Reply:
makes sense